src/irmin-pack/unix/control_file.ml

```diff
@@ -16,114 +16,226 @@
 
 open Import
 include Control_file_intf
+module Version = Irmin_pack.Version
 
-module Plv3 = struct
-  include Payload_v3
-
-  let of_bin_string = Irmin.Type.of_bin_string t |> Irmin.Type.unstage
-end
+module Checksum = struct
+  type 'a target = {
+    encode_bin : 'a -> (string -> unit) -> unit;
+    set_checksum : 'a -> int63 -> 'a;
+    payload : 'a;
+  }
```

Comment:
Nitpick: I find it harder to read the code when a record is defined but turns out to only be consumed by one function (it could be switched to named arguments?)

Reply:
I started with named arguments but then switched to this since it felt like a nice little package for checksum'ing. But! I don't totally love it (naming felt awkward -- what is `target` anyway?), so let me take a pass at pulling this back apart into named args to see how it feels.

Reply:
I just pushed a version that uses named arguments and I feel good with it!